api/base/serializers.py

```diff
@@ -1058,9 +1058,24 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class PrefetchRelationshipsSerializer(ser.Serializer):
+class SparseFieldsetMixin(object):
+    def parse_sparse_fields(self, allow_unsafe=False, **kwargs):
+        request = kwargs.get('context', {}).get('request', None)
+        if request and (allow_unsafe or request.method in permissions.SAFE_METHODS):
+            sparse_fieldset_query_param = 'fields[{}]'.format(self.Meta.type_)
+            if sparse_fieldset_query_param in request.query_params:
+                fieldset = request.query_params[sparse_fieldset_query_param].split(',')
+                for field_name in self.fields.fields.copy().keys():
+                    if field_name in ('id', 'links', 'type'):
+                        # MUST return these fields
+                        continue
+                    if field_name not in fieldset:
+                        self.fields.pop(field_name)
+
+class PrefetchRelationshipsSerializer(ser.Serializer, SparseFieldsetMixin):
```

Comment:
Ah, I see; thanks for the clarification. I think the rename to `BaseAPISerializer` makes sense. Feel free to make that change in this PR, but it's not a blocker.